src/sinol_make/helpers/package_util.py

```diff
@@ -1,12 +1,24 @@
 import os
+import yaml
+import glob
 from enum import Enum
 from typing import List, Union, Dict, Any
 
 from sinol_make import util
 
 
 def get_task_id() -> str:
-    return os.path.split(os.getcwd())[-1]
+    with open(os.path.join(os.getcwd(), "config.yml")) as config_file:
+        config = yaml.load(config_file, Loader=yaml.FullLoader)
+    if "sinol_task_id" in config:
+        return config["sinol_task_id"]
+    else:
+        print(util.warning("sinol_task_id not specified in config.yml. Using task id from directory name."))
+        task_id = os.path.split(os.getcwd())[-1]
+        if len(task_id) == 3:
+            return task_id
+        else:
+            util.exit_with_error("Invalid task id. Task id should be 3 characters long.")
```

Comment:
Do you think we should exit when `sinol_task_id` is not in config or try to get task id from directory's name?

Reply:
I want to avoid heuristics as much as possible. I would say that we can try to read the directory name and:
- if it's 3 letters long, assume it's the task id,
- otherwise make an error that the field is missing.

It would be an undocumented feature, but hopefully that will make sinol-make work with old packages.

What do you think?

Reply:
I agree, that's a good solution